dev/RadioButtons/ColumnMajorUniformToLargestGridLayout.cpp

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "Utils.h"
+#include "ColumnMajorUniformToLargestGridLayout.h"
+
+CppWinRTActivatableClassWithBasicFactory(ColumnMajorUniformToLargestGridLayout);
+
+winrt::Size ColumnMajorUniformToLargestGridLayout::MeasureOverride(
+    winrt::NonVirtualizingLayoutContext const& context,
+    winrt::Size const& availableSize)
+{
+    if (auto const children = context.Children())
+    {
+        auto const maxColumns = std::max(1, MaximumColumns());
+        MUX_ASSERT(maxColumns > 0);
+        auto const maxItemsPerColumn = static_cast<int>(std::ceil(static_cast<double>(children.Size()) / static_cast<double>(maxColumns)));
+
+        m_largestChildSize = [children, availableSize]()
+        {
+            auto largestChildWidth = 0.0f;
+            auto largestChildHeight = 0.0f;
+            for (auto const child : children)
+            {
+                child.Measure(availableSize);
+                auto const desiredSize = child.DesiredSize();
+                if (desiredSize.Width > largestChildWidth)
+                {
+                    largestChildWidth = desiredSize.Width;
+                }
+                if (desiredSize.Height > largestChildHeight)
+                {
+                    largestChildHeight = desiredSize.Height;
+                }
+            }
+            return winrt::Size(largestChildWidth, largestChildHeight);
+        }();
+
+        auto const actualColumnCount = static_cast<float>(std::min(
```

Comment:
>static_cast<float> [](start = 39, length = 18)

casting up to double and then down to float? Why not just cast both params to float?